libs/SalesforceSDKCore/SalesforceSDKCore/Classes/NativeLoginManagerInternal.swift

```diff
@@ -0,0 +1,204 @@
+///
+/// It's only public to be visible from the obj-c code when the library is compiled as a framework.
+/// See https://developer.apple.com/documentation/swift/importing-swift-into-objective-c#Import-Code-Within-a-Framework-Target
+@objc(SFNativeLoginManagerInternal)
+public class NativeLoginManagerInternal: NSObject, NativeLoginManager {
+    @objc let clientId: String
+    @objc let redirectUri: String
+    @objc let loginUrl: String
+    
+    struct AuthorizationResponse: Codable {
+        let sfdc_community_url: String
+        let sfdc_community_id: String
+        let code: String
+    }
+    
+    @objc internal init(clientId: String, redirectUri: String, loginUrl: String) {
+        self.clientId = clientId
+        self.redirectUri = redirectUri
+        self.loginUrl = loginUrl
+    }
+    
+    public func login(username: String, password: String) async -> NativeLoginResult {
+        let trimmedUsername = username.trimmingCharacters(in: .whitespacesAndNewlines)
+        let trimmedPassword = password.trimmingCharacters(in: .whitespacesAndNewlines)
+        
+        if !isValidUsername(username: trimmedUsername) {
+            return .invalidUsername
+        }
+        
+        if !isValidPassword(username: trimmedUsername, password: trimmedPassword) {
+            return .invalidPassword
+        }
+        
+        guard let creds: Data = "\(trimmedUsername):\(trimmedPassword)".data(using: .utf8) else {
+            SFSDKCoreLogger().e(self.classForCoder, message: "Username or Password contain non-UTF8 characters.")
+            return .invalidCredentials
+        }
+        let encodedCreds = urlSafeBase64Encode(data: creds)
+        let authRequest = RestRequest(method: RestRequest.Method.POST, baseURL: loginUrl, path: kSFOAuthEndPointAuthorize, queryParams: nil)
+        let customHeaders: NSMutableDictionary = [kSFOAuthRequestTypeParamName: kSFOAuthRequestTypeNamedUser,
+                                                  kHttpHeaderContentType: kHttpPostContentType,
+                                                  kSFOAuthAuthorizationTypeParamName: "\(kSFOAuthAuthorizationTypeBasic) \(encodedCreds)"]
+        
+        let codeVerifier = generateCodeVerifier()
+        guard let challenge = generateChallenge(codeVerifier: codeVerifier) else { return .unknownError }
+        let authRequestBody = "\(kSFOAuthResponseType)=\(kSFOAuthCodeCredentialsParamName)&\(kSFOAuthClientId)=\(self.clientId)&\(kSFOAuthRedirectUri)=\(redirectUri)&\(kSFOAuthCodeChallengeParamName)=\(challenge)"
+        authRequest.customHeaders = customHeaders
+        authRequest.setCustomRequestBodyString(authRequestBody, contentType: kHttpPostContentType)
+        authRequest.requiresAuthentication = false
+        authRequest.endpoint = ""
+   
+        // First REST Call - Authorization
+        let authResult = await withCheckedContinuation { continuation in
```

Comment:
`await withCheckedContinuation` is straight up magic 🪄.  

I think we should consider wrapping all of our RestClient APIs so customers get that sweet Swift 5.5 syntax.

Reply:
Update:  This worked flawlessly in Xcode 5.1 but now fails (with no code change) in Xcode 5.2 😡.